### PR TITLE
"System back gesture" explanation

### PR DESCRIPTION
**Pull request by justinmc**
This PR improves the PopScope docs by explaining the definition of "system back gesture".  There was some confusion in https://github.com/flutter/flutter/issues/138624 about the behavior of iOS back gestures that this hopes to clarify.

### CODE REVIEW

File: packages/flutter/lib/src/widgets/pop_scope.dart [hunk 1]
```diff
@@ -10,6 +10,12 @@ import 'routes.dart';
 
 /// Manages system back gestures.
 ///
+/// A system back gesture is a swipe inward from near the edge of the screen on
```
Comment (LongCatIsLooong) [inline packages/flutter/lib/src/widgets/pop_scope.dart:13]: Ah so this widget is a no-op if used on any other platforms other than Android? Maybe should make it to the short description line: "Manages system back gestures on Android." ?

File: packages/flutter/lib/src/widgets/pop_scope.dart [hunk 2]
```diff
@@ -10,6 +10,12 @@ import 'routes.dart';
...
+/// Android. It is recognized by Android before being passed to Flutter. iOS has
+/// a similar gesture, but it is recognized in Flutter by
```
Comment (LongCatIsLooong) [inline packages/flutter/lib/src/widgets/pop_scope.dart:15]: Can you use this widget to control the behavior of CupertinoRouteTransitionMixin then?

**Review by LongCatIsLooong — APPROVED** [review]

File: packages/flutter/lib/src/widgets/pop_scope.dart (hunk 2 above)
Comment (justinmc) [reply packages/flutter/lib/src/widgets/pop_scope.dart:15]: Setting `canPop: false` will disable the CupertinoRouteTransitionMixin  back gesture and its animation altogether. I'll explain this better.

File: packages/flutter/lib/src/widgets/pop_scope.dart (hunk 1 above)
Comment (justinmc) [reply packages/flutter/lib/src/widgets/pop_scope.dart]: I've rearranged this to explain it better too.

**Review by LongCatIsLooong — APPROVED** [review]
The documentation changes lgtm. The ios / android swipe gesture difference seems a bit peculiar to me, is there a reason for that?

**Comment by justinmc** [comment]
On Android, Flutter has to allow Android to detect the gesture and report it to Flutter. The reason is that Android shows some native UI (a back arrow) that is styled in a way that Flutter cannot replicate. Android internally does some fancy stuff with the M3 system colors and there is no API that Flutter can use to get the right colors for that.

On iOS it's just all done by Flutter, so there is a fundamental difference there.

**Pull request merged**